### PR TITLE
Lem issue 115

### LINKED ISSUE #115: Client for the indexer
Add client module, that implements indexer's API

### PR DESCRIPTION
**Pull request by LeMarwin**
close #115 and #118

### CODE REVIEW

File: wallet/ergvein-wallet.cabal [hunk 1]
```diff
@@ -75,6 +78,7 @@ library
     , data-default
     , directory
     , ergvein-common            >= 1.0      && < 1.1
+    , ergvein-index-api
```
Comment (NCrashed) [inline wallet/ergvein-wallet.cabal]: Need version

File: wallet/src/Ergvein/Wallet/Monad/Auth.hs [hunk 2]
```diff
@@ -108,7 +122,7 @@ instance (MonadBaseConstr t m, MonadRetract t m, PlatformNatives) => MonadFrontB
   getPasswordSetEF = asks env'passSetEF
   {-# INLINE getPasswordSetEF #-}
   requestPasssword reqE = do
-    idE <- performEvent $ (liftIO randomIO) <$ reqE
+    idE <- performEvent $ (liftIO getRandom) <$ reqE
```
Comment (NCrashed) [inline wallet/src/Ergvein/Wallet/Monad/Auth.hs]: Brackets around liftIO are redundant

File: wallet/src/Ergvein/Wallet/Monad/Client.hs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  -- * Helper functions
+  -- | Set the number of required confirmations
+  setRequiredUrlNum :: Event t Int -> m ()
+  -- | Get a list of indexer urls. At most N urls, where N is set by setRequiredUrlNum
+  getUrlList :: Event t () -> m (Event t [Text])
+  -- | Add a number of urls to the set of valid urls
+  addUrls :: Event t [Text] -> m ()
+  -- | Remove a number of urls from the set of valid urls
+  invalidateUrls :: Event t [Text] -> m ()
+  -- * Client functions
```
Comment (NCrashed) [inline wallet/src/Ergvein/Wallet/Monad/Client.hs]: Haddock will break here

File: wallet/src/Ergvein/Wallet/Monad/Client.hs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import Data.Text
+import Reflex
+
+import Ergvein.Index.API.Types
+import Ergvein.Wallet.Monad.Base
+
+-- | Implements client logic, requesting multiple indexers and verifying results
+class MonadBaseConstr t m => MonadClient t m | m -> t where
+  -- * Helper functions
```
Comment (NCrashed) [inline wallet/src/Ergvein/Wallet/Monad/Client.hs]: `-- *` Will break haddock

File: wallet/src/Ergvein/Wallet/Monad/Auth.hs (hunk 2 above)
Comment (LeMarwin) [reply wallet/src/Ergvein/Wallet/Monad/Auth.hs]: done in 7dc9b0c75cae355c7eef2ff1d285723d60d0684c

File: wallet/ergvein-wallet.cabal (hunk 1 above)
Comment (LeMarwin) [reply wallet/ergvein-wallet.cabal]: done in 7dc9b0c75cae355c7eef2ff1d285723d60d0684c

File: wallet/src/Ergvein/Wallet/Monad/Client.hs (hunk 4 above)
Comment (NCrashed) [reply wallet/src/Ergvein/Wallet/Monad/Client.hs]: https://haskell-haddock.readthedocs.io/en/latest/markup.html#documentation-structure-examples

Comment (LeMarwin) [reply wallet/src/Ergvein/Wallet/Monad/Client.hs]: closed 48fe828bc3000409cbf67be6d754449c6e3cf698

**Pull request merged**